subprojects/testing-jvm/src/main/java/org/gradle/api/internal/tasks/testing/testng/TestNGSpec.java

```diff
@@ -40,21 +37,36 @@ public class TestNGSpec implements Serializable {
     private final boolean preserveOrder;
     private final boolean groupByInstances;
 
-    public TestNGSpec(TestNGOptions options, DefaultTestFilter filter) {
-        this.defaultSuiteName = options.getSuiteName();
-        this.defaultTestName = options.getTestName();
-        this.parallel = options.getParallel();
-        this.threadCount = options.getThreadCount();
-        this.useDefaultListener = options.getUseDefaultListeners();
-        this.includeGroups = options.getIncludeGroups();
-        this.excludeGroups = options.getExcludeGroups();
-        this.listeners = options.getListeners();
-        this.includedTests = filter.getIncludePatterns();
-        this.excludedTests = filter.getExcludePatterns();
-        this.includedTestsCommandLine = filter.getCommandLineIncludePatterns();
-        this.configFailurePolicy = options.getConfigFailurePolicy();
-        this.preserveOrder = options.getPreserveOrder();
-        this.groupByInstances = options.getGroupByInstances();
+    public TestNGSpec(
+        String defaultSuiteName,
+        String defaultTestName,
+        String parallel,
+        int threadCount,
+        boolean useDefaultListener,
+        Set<String> includeGroups,
+        Set<String> excludeGroups,
+        Set<String> listeners,
+        Set<String> includedTests,
+        Set<String> excludedTests,
+        Set<String> includedTestsCommandLine,
```

Comment:
A simple data class grouping these 3 common test selection args I think would help these constructors.

Reply:
I agree, though IMO the changes I want to make to resolve this don't belong in a PR which mostly moves things around. I try to avoid co-locating PR which move content with PRs which update the code itself. While this PR does include some changes, they are needed for the re-location of the files. 

I have a commit waiting to put out for review which addresses this by moving these three fields into a common data class.